src/app/forms/validators/custom-validators.ts

```diff
@@ -179,6 +179,22 @@ export class CustomValidators {
     };
   };
 
+  static dateNotExceed = (sibling: string, months: number): ValidatorFn => {
+    return (control: AbstractControl): ValidationErrors | null => {
+      if (control?.parent) {
+        const siblingControl = control.parent.get(sibling) as CustomFormControl;
+        const siblingDate = new Date(siblingControl.value);
+        siblingDate.setMonth(siblingDate.getMonth() + months);
+
+        if (new Date(control.value) > siblingDate) {
```

Comment:
```suggestion
        const maxDate = new Date(siblingControl.value);
        maxDate.setMonth(maxDate.getMonth() + months);

        if (new Date(control.value) > maxDate) {
```

Actually now that I think about it doesn't make sense for it to be called `siblingDate` as it's never used in that contect in the code